### PR TITLE
CPE - Add Custom Table Column Quick Action

### PR DESCRIPTION
**Pull request by vadson71**
Quick Action for adding custom columns to a regular adaptive, grid, tree or analytical table

### CODE REVIEW

File: packages/preview-middleware-client/test/unit/adp/quick-actions/fe-v4.test.ts [hunk 1]
```diff
@@ -533,6 +670,198 @@ describe('FE V2 quick actions', () => {
...
+            describe('create table custom column', () => {
+                const testCases = [
+                    { tableType: MDC_TABLE_TYPE, dialog: DialogNames.ADD_FRAGMENT, toString: () => MDC_TABLE_TYPE }
+                    // { tableType: TREE_TABLE_TYPE, dialog: DialogNames.ADD_FRAGMENT, toString: () => TREE_TABLE_TYPE },
+                    // {
+                    //     tableType: ANALYTICAL_TABLE_TYPE,
+                    //     dialog: DialogNames.ADD_FRAGMENT,
+                    //     toString: () => ANALYTICAL_TABLE_TYPE
+                    // },
+                    // { tableType: GRID_TABLE_TYPE, dialog: DialogNames.ADD_FRAGMENT, toString: () => GRID_TABLE_TYPE }
```
Comment (nikmace) [inline packages/preview-middleware-client/test/unit/adp/quick-actions/fe-v4.test.ts]: Is this comment still needed? Or is it WIP?

**Review by nikmace — CHANGES_REQUESTED** [review]
Overall the code looks good, I need to test manually when it is ready
Only some minor comments

File: packages/preview-middleware-client/test/unit/adp/quick-actions/fe-v4.test.ts (hunk 1 above)
Comment (nikmace) [reply packages/preview-middleware-client/test/unit/adp/quick-actions/fe-v4.test.ts]: outdated, already uncommented

File: packages/preview-middleware-client/src/adp/command-executor.ts [hunk 2]
```diff
@@ -52,12 +53,29 @@ export default class CommandExecutor {
...
     }
 
+    /**
+     * Creates composite command without nested commands
+     *
+     * @param runtimeControl Managed object
+     */
+    public async createCompositeCommand(runtimeControl: ManagedObject): Promise<CompositeCommand> {
+        try {
+            return await CommandFactory.getCommandFor<CompositeCommand>(runtimeControl, 'composite');
```
[the remark this answers is not in the document]
Comment (vadson71) [reply packages/preview-middleware-client/src/adp/command-executor.ts:63]: removed

File: packages/adp-tooling/templates/rta/common/analytical-custom-column.xml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+
```
[the remark this answers is not in the document]
Comment (vadson71) [reply packages/adp-tooling/templates/rta/common/analytical-custom-column.xml:1]: removed

File: packages/preview-middleware-client/src/adp/controllers/AddFragment.controller.ts [hunk 4]
```diff
@@ -184,26 +186,8 @@ export default class AddFragment extends BaseDialog<AddFragmentModel> {
...
-        const allAggregations = Object.keys(controlMetadata.getAllAggregations());
-        const hiddenAggregations = ['customData', 'layoutData', 'dependents'];
-        const targetAggregation = allAggregations.filter((item) => {
-            if (hiddenAggregations.indexOf(item) === -1) {
-                return item;
-            }
-            return false;
-        });
+        const metadata = this.getControlMetadata();
+        const { controlMetadata, targetAggregation } = metadata;
```
[the remark this answers is not in the document]
Comment (vadson71) [reply packages/preview-middleware-client/src/adp/controllers/AddFragment.controller.ts:190]: changed

File: packages/preview-middleware-client/src/adp/controllers/AddTableColumnFragments.controller.ts [hunk 5]
```diff
@@ -0,0 +1,325 @@
...
+        }
+    }
+
+    /**
+     * Builds data that is used in the dialog
+     */
+    async buildDialogData(): Promise<void> {
+        const metadata = this.getControlMetadata();
+
+        const { controlMetadata } = metadata;
```
[the remark this answers is not in the document]
Comment (vadson71) [reply packages/preview-middleware-client/src/adp/controllers/AddTableColumnFragments.controller.ts:166]: changed

File: packages/preview-middleware-client/src/adp/controllers/AddTableColumnFragments.controller.ts [hunk 6]
```diff
@@ -0,0 +1,325 @@
...
+                fragmentPath: `fragments/${fragment.fragmentName}.fragment.xml`,
+                index: index ?? 0,
+                targetAggregation:
+                    fragment.targetAggregation === ITEMS_AGGREGATION ? CELLS_AGGREGATION : fragment.targetAggregation
+            };
+
+            const command = await this.commandExecutor.getCommand<AddTableCellFragmentChangeContentType>(
+                fragment.targetAggregation === COLUMNS_AGGREGATION
+                    ? this.runtimeControl
+                    : (this.runtimeControl.getAggregation(ITEMS_AGGREGATION) as ManagedObject[])[0],
```
[the remark this answers is not in the document]
Comment (vadson71) [reply packages/preview-middleware-client/src/adp/controllers/AddTableColumnFragments.controller.ts:303]: done

File: packages/preview-middleware-client/src/adp/controllers/AddTableColumnFragments.controller.ts [hunk 7]
```diff
@@ -0,0 +1,325 @@
...
+                modifiedValue,
+                designMetadata,
+                flexSettings
+            );
+
+            const templateName =
+                fragment.targetAggregation === COLUMNS_AGGREGATION ? `V2_SMART_TABLE_COLUMN` : 'V2_SMART_TABLE_CELL';
+            const preparedChange = command.getPreparedChange();
+            const content = { ...preparedChange.getContent() };
+            content.templateName = templateName;
```
[the remark this answers is not in the document]
Comment (vadson71) [reply packages/preview-middleware-client/src/adp/controllers/AddTableColumnFragments.controller.ts:314]: done

**Review by nikmace — APPROVED** [review]
Code looks good
Comments addressed
Coverage is good
_Did not test manually_

**Review by voicis — APPROVED** [review]
- Code changes looks good
- Good test coverage
- Did not test locally

**Pull request merged**